src/startup.js

```diff
@@ -1,132 +1,149 @@
 
-		// Read the env now because we can't overwrite everything yet
+			// Get options from the script tag
+			if (isWebWorker) {
+				resolve({
+					stealURL: location.href
+				});
 
-		// immediate steals we do
-		var steals = [];
+			} else if(isBrowserWithWindow) {
+				// if the browser supports currentScript, us it!
+				if (document.currentScript) {
+					resolve(getScriptOptions(document.currentScript));
+					return;
+				}
 
-		// we only load things with force = true
-		if ( System.loadBundles ) {
+				// dealing with async & deferred scripts
+				// set an onload handler for all script tags and the first one which executes
+				// is your stealjs
+				var scripts = document.scripts;
+				function onLoad() {
+					for (var i = 0; i < scripts.length; ++i) {
+						scripts[i].removeEventListener('load', onLoad, false);
+					}
+					resolve(getScriptOptions(event.target));
+				}
+				for (var i = 0; i < scripts.length; ++i) {
+					scripts[i].addEventListener('load', onLoad, false);
+				}
 
-			if(!System.main && System.isEnv("production") && !System.stealBundled) {
-				// prevent this warning from being removed by Uglify
-				var warn = console && console.warn || function() {};
-				warn.call(console, "Attribute 'main' is required in production environment. Please add it to the script tag.");
+			} else {
+				// or the only option is where steal is.
+				resolve({
+					stealPath: __dirname
+				});
 			}
+		})
+	};
 
-			configDeferred = System["import"](System.configMain);
+	steal.startup = function(config){
+		appPromise = getUrlOptions().then(function(urlOptions) {
 
-			appDeferred = configDeferred.then(function(cfg){
-				setEnvsConfig.call(System);
-				return System.main ? System["import"](System.main) : cfg;
-			});
+			if (typeof config === 'object') {
+				// the url options are the source of truth
+				config = extend(config, urlOptions);
+			} else {
+				config = urlOptions;
+			}
 
-		} else {
-			configDeferred = System["import"](System.configMain);
+			// set the config
+			System.config(config);
 
-			devDeferred = configDeferred.then(function(){
-				setEnvsConfig.call(System);
-				setupLiveReload.call(System);
+			setEnvsConfig.call(this.System);
 
-				// If a configuration was passed to startup we'll use that to overwrite
-				// what was loaded in stealconfig.js
-				// This means we call it twice, but that's ok
-				if(config) {
-					System.config(config);
-				}
+			// Read the env now because we can't overwrite everything yet
 
-				return System["import"]("@dev");
-			});
+			// immediate steals we do
+			var steals = [];
 
-			appDeferred = devDeferred.then(function(){
-				// if there's a main, get it, otherwise, we are just loading
-				// the config.
-				if(!System.main || System.env === "build") {
-					return configDeferred;
-				}
-				var main = System.main;
-				if(typeof main === "string") {
-					main = [main];
-				}
-				return Promise.all( map(main,function(main){
```

Comment:
This was to support multi-mains (where main is an array), did you remove support for this on purpose?